hasura/graphql-engine
# console: respect read-only mode in actions pages (close #4656)

### Description

Actions in Hasura  console will now respond to read only mode. This will hide/disable components when read only mode is enabled in the global state.




### Changelog


### Affected components



### Related Issues



close #4656

### Solution and Design


Referring to Data and Remote Schema, read only mode is implemented. 
Solution will 

- Hide components if the action is just a button or navigation components (Edit, Create, Remove Buttons)
- When there an action components which also shows information, component will be disabled but the information will still show up. (Editors, text area, input etc.)

### Steps to test and verify



1. Go to Hasura Console in read only mode (for testing you may have to set it in the [main](https://github.com/hasura/graphql-engine/blob/ff3ac78d8f11692b939ebb7a221ea1ae73930d4c/console/src/components/Main/State.js#L5) redux state )
2. Go to Actions
3. Create Table buttons from both sidebar and landing page should not be visible.
4. Select any existing actions (create without read only mode if there is nothing)
5. Modify
   - All inputs should be disabled.
   - Buttons should be hidden.
6. Relationships
   - Edit & Add Relationships button should be disabled.
&. Permissions
   - No Create/Edit operations should work (Similar to Data module)
